core/src/main/scala/pipeflow/system/actors/PeriodicSchedulerActor.scala

```diff
@@ -0,0 +1,89 @@
+package pipeflow.system.actors
+
+import java.time.{Clock, OffsetDateTime}
+import java.util.concurrent.TimeUnit
+
+import akka.actor.{Actor, ActorRef, PoisonPill, Props}
+import akka.event.LoggingReceive
+import akka.pattern.pipe
+import pipeflow.dsl.nodes.Node
+import pipeflow.system.PipeFlowSystem.NodeBuilder
+import pipeflow.system.iso8601.{IntervalDuration, RepeatingInterval}
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ExecutionContext, Future}
+
+object PeriodicSchedulerActor {
+
+  case class Tick(recurrences: Option[Long], scheduledDateTime: OffsetDateTime)
+
+  case class NodeCreated(node: Node)
+
+  def props(taskSchedulerRef: ActorRef, repeatingInterval: RepeatingInterval, nodeBuilder: NodeBuilder)(
+            implicit clock: Clock, executionContext: ExecutionContext): Props =
+    Props(new PeriodicSchedulerActor(taskSchedulerRef, repeatingInterval, nodeBuilder))
+}
+
+class PeriodicSchedulerActor(taskSchedulerRef: ActorRef,
+                             repeatingInterval: RepeatingInterval,
+                             nodeBuilder: NodeBuilder)(
+                             implicit clock: Clock,
+                             executionContext: ExecutionContext) extends Actor {
+
+  import PeriodicSchedulerActor.{NodeCreated, Tick}
+
+  override def preStart(): Unit = {
+    if (repeatingInterval.recurrences.forall(_ > 0)) {
+      val now = OffsetDateTime.now(clock)
+      val start = repeatingInterval.start.getOrElse(now)
+      val nextDateTime = calculateNextDateTime(now, start, repeatingInterval.duration)
+      scheduleTick(repeatingInterval.recurrences, nextDateTime, nextTickDuration(now, nextDateTime))
+    }
+    else {
+      self ! PoisonPill
+    }
+  }
+
+  def receive = LoggingReceive {
+    case Tick(recurrences, scheduledDateTime) =>
+      Future {
+        val node = nodeBuilder(scheduledDateTime.toLocalDateTime)
+        NodeCreated(node)
+      }.pipeTo(taskSchedulerRef)
+
+      val now = OffsetDateTime.now(clock)
+      val nextRecurrences = recurrences.map(_ - 1)
+      if (nextRecurrences.forall(_ > 0)) {
```

Comment:
`forall` returns true if the option is empty, is that what you need or `exist` is what u after?

Reply:
Yes, that's exactly what is needed here, either the recurrences have not been specified (which means run infinitely), or else the value is greater than 0.